include/swift/AST/DiagnosticsSema.def

```diff
@@ -1294,8 +1294,8 @@ ERROR(nominal_type_not_attribute,none,
 
 ERROR(mutating_invalid_global_scope,none, "%0 is only valid on methods",
       (SelfAccessKind))
-ERROR(mutating_invalid_classes,none, "%0 isn't valid on methods in "
-      "classes or class-bound protocols", (SelfAccessKind))
+ERROR(mutating_invalid_classes,none, "%0 isn't valid on %select{methods|accessors}1 in "
+      "classes or class-bound protocols", (SelfAccessKind, bool))
```

Comment:
While we're here, should we `%select{classes|class-bound protocols}` too?

Reply:
Yeah we could do that! I think it would've been nice if we had something like `DescriptiveContextKind` which could return diagnostic-friendly context name (like "local/nested method", "extension of class-bound protocol", "initializer", etc) so we could cover more scenarios (not just here but in other places as well where we use hardcoded names).

Reply:
May we get rid of the contraction in this message too?

Reply:
Sure, done!